internal/statesync/stateprovider.go

```diff
@@ -33,20 +39,17 @@ type StateProvider interface {
 	State(ctx context.Context, height uint64) (sm.State, error)
 }
 
-// lightClientStateProvider is a state provider using the light client.
-type lightClientStateProvider struct {
+type stateProviderRPC struct {
 	tmsync.Mutex  // light.Client is not concurrency-safe
 	lc            *light.Client
-	version       sm.Version
 	initialHeight int64
 	providers     map[lightprovider.Provider]string
```

Comment:
Optional refactor: The p2p and rpc provider appear to share a good amount of code. We can achieve some nicer code reuse by changing how the state providers are constructed. It appears that the `stateProviderRPC` just relies on its `providers` for fetching consensus parameters and doesn't need them beyond the constructor.  Similarly the `stateProviderP2P` holds a pair of channels for retrieving the consensus parameters and adds the providers to the light client it contains. 

A simple refactor to make both of these types of state provider share a type may make this slightly more reuseable. Namely, we could update the the state provider to hold a `paramsFetcher` and implement two`paramsFetcher`s: one using p2p and one using RPC.

Reply:
I definitely think there's a lot of duplication here that can be cleaned up. I think to start off with I just wanted to copy them because I wasn't sure how much they would change